contrib/fslhd.py

```diff
@@ -257,8 +260,39 @@ def generate_nifti_fields(header):
         'sform_xorient': ORIENTATION_STRINGS[nib.aff2axcodes(sform)[0]],
         'sform_yorient': ORIENTATION_STRINGS[nib.aff2axcodes(sform)[1]],
         'sform_zorient': ORIENTATION_STRINGS[nib.aff2axcodes(sform)[2]],
-        'file_type': f"NIFTI-{int(nib_fields['magic'].decode('utf-8')[2])}+",
+        'file_type': f"NIFTI-{int(nib_fields['magic'].decode('utf-8', errors='ignore')[2])}+",
         'file_code': int(nib_fields['magic'].decode("utf-8")[2]),
-        'descrip': nib_fields['descrip'].decode('utf-8'),
-        'aux_file': nib_fields['aux_file'].decode('utf-8')
+        'descrip': nib_fields['descrip'].decode("utf-8", errors="ignore"),
+        'aux_file': nib_fields['aux_file'].decode("utf-8", errors="ignore")
+    }
+
+
+def generate_numpy_fields(arr):
+    """
+    Generate useful summary statistics for any arbitrary numpy array.
+    """
+    arr = arr.astype(np.float64)
+    return {
+        'arr_nbytes': arr.nbytes,
+        'arr_size': arr.size,
+        'arr_num_zeros': arr[arr == 0.0].size,
+        'arr_num_nonzeros': arr[arr != 0.0].size,
+        'arr_num_positives': arr[arr > 0.0].size,
+        'arr_num_negatives': arr[arr < 0.0].size,
+        'arr_num_nans': arr[arr == np.nan].size,
+        'arr_num_infs': arr[arr == np.inf].size,
```

Comment:
One weird corner of IEEE floating point numbers is that `np.nan != np.nan`, so `arr_num_nans` will always be zero. There's a specific function to test whether a floating point value is `nan` (and also one for (possibly negative) infinity):
```suggestion
        'arr_num_nans': np.count_nonzero(np.isnan(arr)),
        'arr_num_infs': np.count_nonzero(np.isinf(arr)),
```